### PR TITLE
feat(ctb): spacer validation task

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Introduces a new task for validating the correctness of spacer variables. Prevents future developers from accidentally moving spacers.

### CODE REVIEW

File: packages/contracts-bedrock/tasks/validate-spacers.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            const [, slot, offset] = variable.label.split('_')
+            if (slot !== variable.slot) {
+              throw new Error(
+                `${contractName} ${variable.label} is in slot ${variable.slot} but should be in ${slot}`
+              )
+            }
+            if (parseInt(offset, 10) !== variable.offset) {
+              throw new Error(
+                `${contractName} ${variable.label} is at offset ${variable.offset} but should be at ${offset}`
+              )
```
Comment (maurelian) [inline packages/contracts-bedrock/tasks/validate-spacers.ts:30]: I noticed that you're not checking that the length given in the space name matches the type...
which could be done easily enough given all spacers are currently either a mapping, uint or address.

That would be a slight improvement, but I'm fine either way. Will approve now and you can resolve the comment if you just want to merge it.

**Review by maurelian — APPROVED** [review]

File: packages/contracts-bedrock/tasks/validate-spacers.ts (hunk 1 above)
Comment (smartcontracts) [reply packages/contracts-bedrock/tasks/validate-spacers.ts:34]: Oops, good catch. Added an simple version of this to the script.

**Comment by maurelian** [comment]
@mergify bump

**Review by tynes — APPROVED** [review]
Easy to read and understand

**Pull request merged**